pkg/util/log/log.go

```diff
@@ -618,6 +645,16 @@ func Warnf(format string, params ...interface{}) error {
 	return logFormatWithError(seelog.WarnLvl, func() { Warnf(format, params...) }, Logger.warnf, format, false, params...)
 }
 
+// WarnfStackDepth logs with format at the warn level and the current stack depth plus the given depth
+func WarnfStackDepth(format string, depth int, params ...interface{}) error {
+	currentLevel, _ := GetLogLevel()
+	if currentLevel <= seelog.WarnLvl {
+		m := fmt.Sprintf(format, params...)
+		return logContextWithError(seelog.WarnLvl, func() { Warnc(m, nil) }, Logger.warn, m, false, depth, nil)
+	}
+	return nil
+}
```

Comment:
For `Warnf` and `Errorf` we always have to return the real error (some code do `return log.Error(...)`), so these two actually shouldn't check the log level.

See how the equivalent are defined in the log component: https://github.com/DataDog/datadog-agent/blob/443ac501027aecbc0cc754efc8055daaef2eedc3/comp/core/log/logger.go#L116-L120

Reply:
Ah, good catch! Thanks :)